### ISSUE TITLE
go-critic intermittently not catching errors

### ISSUE THREAD

**Issue by ihgann** [go-critic/go-critic#715]
`go-critic` version: `0.3.3`
`go-critic` binary hash (md5): `c5d9ba8922d922f0d334013b727799d0`

My operating system is macOS High Sierra version 10.13.6.

**Description**

Intermittently, `go-critic` does not catch errors. Out of 10 runs, only 3/10 catch errors and run a failure.

I'm running with

```bash
${GOPATH}/bin/gocritic check-project -withOpinionated -withExperimental .
```

Specifically, it's *occasionally* not catching on the `commentedOutCode` experimental checker, although I've yet to confirm if any other checkers are intermittently failing (so it's possible it's just because it's experimental).

The failure catches the following code line:

```golang
// cfg.Ignore = ...
```

so the failure is valid, with the failure response:

```bash
/code/go/bin/gocritic check-project -withOpinionated -withExperimental .
$GOPATH/src/create-go-app/internal/create-go-app/app/scripts/build_assets.go:79:2: commentedOutCode: may want to remove commented-out code
lint error: exit status 1
```

**Comment by Quasilyte** [comment]
Interesting. It might be related with recent switch to `go/packages` from the `loader`. (#697)

Could you provide all code required to do a reproducer?

**Comment by Quasilyte** [comment]
Or, if it's simpler for you, maybe you can check whether this issue is still reproducible using gocritic built from the d9a331cd6632bc7e1549a7714f1c594a4f65f19c?

**Comment by ihgann** [comment]
@Quasilyte - I verified with top-of-tree, as well as [d9a331c](https://github.com/go-critic/go-critic/commit/d9a331cd6632bc7e1549a7714f1c594a4f65f19c) - in both cases it consistently failed on *something*, but would intermittently **not** catch *all* errors.

For example, I have two files in two different directories (both under `internal` if that happens to make any difference):

```
x := "test"
// x = "commented"
fmt.Println(x)
```

The same code in both the files: `./internal/app/app.go` and `./internal/app/scripts/script.go`

Now, with the build from this commit (as well as top-of-tree, `858f8b2fb7bac1f9bca50c919a292009e2042430`), I consistently fail on `commentedOutCode` checker for the file `./internal/app/app.

~20% of the time it will also catch `./internal/app/scripts/script.go`, but the other 80% of the time it ignores it.

What's interesting, however, is if I remove the commented line from `./internal/app/app.go`, such that the code is simply:

```
x := "test"
fmt.Println(x)
```

then it goes back to intermittently succeeding, ignoring `./internal/app/scripts/script.go` (just like the original bug report).

--

If it helps further, locally I'm using `go1.11.2 darwin/amd64`. I've tested with each version of `go1.11`, however, and each has produced the same result.

**Comment by Quasilyte** [comment]
I reproduced this issue in a small project.
Will write back or send a fix when I'll understand why it happens.

**Comment by Quasilyte** [comment]
Should be fixed in a new gocritic.
Install from sources:
```bash
go get -v github.com/go-lintpack/lintpack/...
go get -v github.com/go-critic/go-critic/...
lintpack build -o $GOPATH/bin/gocritic -linter.name='gocritic' github.com/go-critic/go-critic/checkers
```

**Comment by Quasilyte** [comment]
For most users, running gocritic from golangci-lint is a more convenient and reliable way.
Closing this issue for now.
If the problem continues to exist, it can be re-opened.
